recipes-core/disk-encryption/disk-encryption.bb

```diff
@@ -36,9 +43,14 @@ do_install() {
     install -d ${D}${sysconfdir}/init.d
     install -m 0755 ${WORKDIR}/init ${D}${sysconfdir}/init.d/disk-encryption
 
-    # Create environment file for TARGET_LUN
-    install -d ${D}${sysconfdir}/default
-    echo "TARGET_LUN=${TARGET_LUN}" > ${D}${sysconfdir}/default/disk-encryption
+    # Create environment file for init service
+    install -d ${D}${sysconfdir}/default ${D}${sysconfdir}/disk-encryption
+    echo "TARGET_LUN=${TARGET_LUN}
+    DISK_ENCRYPTION_KEY_STORAGE=${DISK_ENCRYPTION_KEY_STORAGE}" > ${D}${sysconfdir}/default/disk-encryption
+    if [ "$DISK_ENCRYPTION_KEY_STORAGE" = "file" ]; then
+        echo "DISK_ENCRYPTION_KEY_FILE=/etc/disk-encryption/key" >> ${D}${sysconfdir}/default/disk-encryption
+        install -m 0644 ${WORKDIR}/disk-encryption-key.mustache ${D}${sysconfdir}/disk-encryption/key.mustache
+    fi
 }
 
-FILES:${PN} += "${sysconfdir}/default/disk-encryption"
+FILES:${PN} += "${sysconfdir}/default/disk-encryption ${D}${sysconfdir}/disk-encryption/key.mustache"
```

Comment:
this might fail the build if the DISK_ENCRYPTION_KEY_STORAGE is not equal to file
You can append this also conditionally. Yocto recipes has special syntax for that. Something that looks like this
FILES:${PN}:append = ${@bb.utils.contains(' .....)}